functions/src/modules/SE.functions.ts

```diff
@@ -88,3 +93,230 @@ export const processNewAllotment = functions.database
     });
     return 1;
   });
+
+
+export const importChunks = functions.runWith({
+    timeoutSeconds: 300,
+    memory: '1GB'
+}).https.onRequest( async (req, res) => {
+    const auth = await google.auth.getClient({
+        scopes: ["https://www.googleapis.com/auth/spreadsheets.readonly"]
+    });
+
+    const sheets = google.sheets({ version: "v4", auth });
+    const spreadsheetId = functions.config().reporting.content.processing.spreadsheet_id;
+    let { sheetTitles, columnsIndex } = await helpers.buildSheetIndex(sheets, spreadsheetId);
+
+    let resolutions = ['OK', 'Drop', 'Duplicate', 'On hold', 'Reallot', 'Repeat', 'Derivative'];
+    let resolution, fidCheckRes; // variables to hold values of Res and fidRes when looping through the rows
+
+
+    let lastFile, // used to increment the chunks of an audiofile coming from multiple rows into a single object
+        summary = {
+            addedChunks: 0, 
+            emptyValues: 0, 
+            misplaced: 0, 
+            noResolution: 0, // rows with No `Resolution` or No `FidelityCheckResolution` values
+            overlappingChunks: [], // Chunks with overlapping times
+            modified: [] // Rows already found in the DB
+        };
+
+    
+    for (let i = 0; i < sheetTitles.length; i++) { // Loop throughout the sheets of interest
+
+        let rows = await helpers.getSheetRows(sheetTitles[i], sheets, spreadsheetId);
+    
+        lastFile = { 
+            file_name: null,
+            chunks: [], 
+            skip: false, 
+            track: null 
+        };
+
+        ////////////////////////////////////////
+        //  
+        //  1. loop through all the rows in the current sheet
+        //  2. Apply the following checks and if something wrong is found
+        //      increment the corresonding error attribute in the `summary` object
+        //      [CHECKS]
+        //          A. Misplaced files
+        //          B. Non-existence of `Resolution` values in a row
+        //          C. Empty values in either `Beginning` or `Ending` attributes
+        //
+        //
+        ////////////////////////////////////////
+        for (const row of rows) {
+            let skip = false;
+
+
+            // [CHECK #1] Misplaced audio file entry            
+            if (helpers.extractListFromFilename(row[columnsIndex['AudioFileName']]) != sheetTitles[i].title) {
+                summary.misplaced++;
+                continue;
+            }
+
+
+            resolution = row[columnsIndex['Resolution']];
+            fidCheckRes = row[columnsIndex['FidelityCheckResolution']];
+
+            if (resolution && fidCheckRes) {
+                // [CHECK #2] Non-existence of `Resolution` values in a row
+                if (resolutions.indexOf(resolution) < 0 || resolutions.indexOf(fidCheckRes) < 0 || resolution != fidCheckRes) {
```

Comment:
1. Don't mix two validity checks in one place, as there should be different messages in the summary for them.
1. As far as I know, `indexOf` is case-sensitive, whereas the requirement is that the resolutions should be compared case-insensitively.
1. Here you just skip the row, whereas the validity check failure should produce a warning in the log and summary email.